### DISCUSSION TITLE
Trying to get Mealie to connect to Ollama

### DISCUSSION THREAD

**Discussion by Unibrowser-github** [mealie-recipes/mealie#4766] [Q&A]
Hello, im new to mealie but so far my friends and family love it. But I am having a bit of trouble getting mealie to connect to Ollama docker container properly. When I up load an image of a recipe the ollama log shows the following

  File "/app/mealie/services/recipe/recipe_service.py", line 497, in build_recipe_from_images
    raise Exception("Failed to call OpenAI services") from e
Exception: Failed to call OpenAI services

I have searched the internet for a walkthrough on how to set this up properly but have yet to find anything. Any help would be appreciated.

**Comment by michael-genson** [comment]
Hey there!

Our docs have some instructions on setting up with Ollama: https://docs.mealie.io/documentation/getting-started/installation/open-ai/#configuration

While Ollama can work, users have generally reported that it's not very effective due to hardware limitations, so YMMV. I also don't think users have been successful with image recognition, so you may consider turning that off if you can't get it to work ([see the backend config](https://docs.mealie.io/documentation/getting-started/installation/backend-config/#openai))

**Reply by Unibrowser-github** [reply]
> Hey there!
> 
> Our docs have some instructions on setting up with Ollama: https://docs.mealie.io/documentation/getting-started/installation/open-ai/#configuration
> 
> While Ollama can work, users have generally reported that it's not very effective due to hardware limitations, so YMMV. I also don't think users have been successful with image recognition, so you may consider turning that off if you can't get it to work ([see the backend config](https://docs.mealie.io/documentation/getting-started/installation/backend-config/#openai))

I have read through all of that but idk what i'm doing wrong. I have some very powerful hardware so i'm not sure that will be an issue. So the image recognition really only works properly on GPT currently? Thats a shame. I did see there was a Ollama vision model called LLAVA  i think it was. Maybe someone could figure it out and make a guide. Would be cool!

**Reply by michael-genson** [reply]
There's nothing stopping you on the Mealie side from getting locally-hosted image-capable models working, it's just a hardware limitation for 99% of users.

> I have some very powerful hardware

LLAVA needs something like 48GB of VRAM to work well with a large enough context window for any real applications, including Mealie. You can get it working with less, but the results aren't great.

**Reply by michael-genson** [reply]
FYI most users have found the default context window not to work with Mealie, so if you're not setting it higher [you might want to bump it up](https://github.com/ollama/ollama/blob/main/docs/faq.md#how-can-i-specify-the-context-window-size)
